File: Problems/lc_14.c


char * longestCommonPrefix(char ** strs, int size)
{
    int i;
    int n;
    char c;
    bool found;

    if (size == 1)
        return strs[0];
    if (size == 0)
        return "";
    i = 0;
    while (42)
    {
        n = 0;
        if (strs[0][i] == '\0')
            break ;
        c = strs[0][i];
        found = false;
        while (n < size)
        {
            if (strs[n][i] == '\0' || strs[n][i] != c)
            {
                found = true;
                break ;
            }
            n++;
        }
        if (found == true)
            break ;
        i++;
    }
    printf("%d\n", i);
    strs[0][i] = '\0';
    return  strs[0];
}

File: Py_Session/lc_14.py
import itertools

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""
        for i, tup in enumerate(itertools.zip_longest(*strs)):
            if len(set(tup)) > 1:
                return strs[0][:i]
        return strs[0]
